plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -237,8 +248,25 @@ func init() {
 	})
 }
 
-func createRealIngestor(client localClient, database string, tableName string) (localIngestor, error) {
-	ingestor, err := ingest.New(client.(*kusto.Client), database, tableName, ingest.WithStaticBuffer(bufferSize, maxBuffers))
+// For each table create the ingestor
+func createIngestorByTable(client *kusto.Client, database string, tableName string, ingestionType string) (ingest.Ingestor, error) {
+	var ingestor ingest.Ingestor
+	var err error
+	if strings.ToLower(ingestionType) == managedIngestion {
+		mi, err := ingest.NewManaged(client, database, tableName)
+		if err != nil {
+			return nil, err
+		}
+		ingestor = mi
+	} else if strings.ToLower(ingestionType) == queuedIngestion {
+		qi, err := ingest.New(client, database, tableName, ingest.WithStaticBuffer(bufferSize, maxBuffers))
+		if err != nil {
+			return nil, err
+		}
+		ingestor = qi
+	} else {
+		err = errors.New(`ingestion_type has to be one of managed or queued`)
+	}
```

Comment:
Please use a switch statement here in order to allow easy extension in the future.